client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/HotrodBasedDeviceConnectionInfo.java

```diff
@@ -437,4 +484,17 @@ private void checkForCacheAvailability(final Promise<Status> status) {
     @Override
     public void registerLivenessChecks(final HealthCheckHandler livenessHandler) {
     }
+
+    private Span createSpan(final String operationName, final String tenantId, final String deviceId,
+            final String adapterInstanceId, final SpanContext spanContext) {
+        final Tracer.SpanBuilder spanBuilder = TracingHelper.buildChildSpan(tracer, spanContext, operationName, getClass().getSimpleName())
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
```

Comment:
IMHO this is a client as we are *retrieving* data from the remote grid, aren't we?

Reply:
Yes, right. I've changed it to `SPAN_KIND_CONSUMER`.

Reply:
FMPOV it should be `SPAN_KIND_CLIENT` as we invoke a remote API and get back a result (the data from the grid) ...

Reply:
Yes, fits better. I've changed it accordingly.